tests/unit/components/CombinationMenuItem-test.js

```diff
@@ -0,0 +1,237 @@
+        }.bind(vm)
+
+        wrapper.find('span').trigger('click')
+
+        // assertions
+        // expect(push.firstCall.args[0].name).to.equal("workbenchAddress")
+        // expect(push.firstCall.args[0].params).to.include({ scrollID })
+        // expect(push.firstCall.args[0].params).to.include({ scrollVersionID })
+    })
+
+    it('can handle $post return data when fetching children', done => {
+        const returnData = {
+            status: 200,
+            data: {
+                results: [],
+            },
+        }
+
+        // create wrapper with mocked route and routers
+        let wrapper = mount(CombinationMenuItem, {
+            propsData: {
+                menuType: menuType,
+            },
+        })
+        wrapper.vm.$post = function() {
+            // adhere to interface
+            return new Promise((resolve, reject) => resolve(returnData))
+        }.bind(vm)
+
+        wrapper.vm.fetchChildren()
+
+        // assertions
+        wrapper.vm.$nextTick(() => {
+            // How do I check my data() to make sure this.children is correct?
+            // expect(wrapper.vm.children).to.equal([])
+            done() // ends the test
+          })
+    })
+
+    it('can clone a scroll', done => {
+        const returnData = {
+            status: 200,
+            data: {
+                results: [],
+            },
+        }
+
+        // create wrapper with mocked route and routers
+        let wrapper = mount(CombinationMenuItem, {
+            propsData: {
+                menuType: menuType,
+            },
+        })
+        wrapper.vm.$post = function() {
+            // adhere to interface
+            return new Promise((resolve, reject) => resolve(returnData))
+        }.bind(vm)
+
+        wrapper.vm.cloneScroll()
+
+        // assertions
+        wrapper.vm.$nextTick(() => {
+            // How do I check my data() to make sure this.children is correct?
+            // expect(wrapper.vm.children).to.equal([])
+            done() // ends the test
+          })
+    })
+
+    it('watches for menuType changes', done => {
+        const returnData = {
+            status: 200,
+            data: {
+                results: [],
+            },
+        }
+
+        // create wrapper with mocked route and routers
+        let wrapper = mount(CombinationMenuItem, {
+            propsData: {
+                menuType: menuType,
+            },
+        })
+        wrapper.vm.$post = function() {
+            // adhere to interface
+            return new Promise((resolve, reject) => resolve(returnData))
+        }.bind(vm)
+
+        // const spy = sinon.spy(wrapper.vm.fetchChildren)
+        wrapper.vm.open = true
+        wrapper.setProps({menuType: 'image'})
+        
+
+        // assertions
+        wrapper.vm.$nextTick(() => {
+            // How do I check my data() to make sure this.children is correct?
+            // expect(wrapper.vm.children).to.equal([])
```

Comment:
`vm.children` should be right. Try a breakpoint here in Chrome debugger and see if you can inspect the contents.

Reply:
Yeah, I ran the debugger over this, and it was rather surprising.  When I created that local variable in fetchChildren() and watched it, I could see in the debugger that it correctly received the response I spoofed in the $post function redefinition of the unit test.  But when I switch back to using the variable as defined in data() the response seemed to go into a black whole and the value of wrapper.vm.children was always [] (the initial value as set in data() ).  Very strange indeed.